### PR TITLE
Staff part

### PR DESCRIPTION
**Pull request by DmitryPateew**
24. Add view, edit, and add mode to the staff page.
Updated navigation

### CODE REVIEW

File: src/components/staffPage/staffPage.tsx [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+      minWidth: 120,
+    },
+    selectEmpty: {
+      marginTop: theme.spacing(2),
+    },
+  }),
+);
+
+interface IUrl {
+  add: string | undefined;
```
Comment (ZmicerBoksha) [inline src/components/staffPage/staffPage.tsx]:
```suggestion
  add?: string;
```

File: src/components/staffPage/staffPage.tsx [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+          {window.location.href.split('/').slice(-2)[0]} page
+        </Typography>
+        <FormControlLabel
+          control={<Switch checked={edit} onChange={() => setEdit(!edit)} name="checkedA" color="primary" />}
+          label=""
+        />
+        <Typography variant="h6" className="switchLabel" noWrap>
+          Edit mode
+        </Typography>
+        {edit ? <Button variant="contained">Save</Button> : ''}
```
Comment (ZmicerBoksha) [inline src/components/staffPage/staffPage.tsx]:
```suggestion
        {edit && <Button variant="contained">Save</Button> }
```

File: src/components/staffPage/staffPage.tsx [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+                    />
+                  );
+                }}
+              />
+            </Paper>
+          </Grid>
+          <input type="submit" />
+        </Grid>
+      </form>
+      {addMode ? '' : <CandidateTrello timeZon={staffData?.empTimezone} />}
```
Comment (ZmicerBoksha) [inline src/components/staffPage/staffPage.tsx]:
```suggestion
      {!addMode && <CandidateTrello timeZon={staffData?.empTimezone} />}
```

File: src/components/table/table.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    state: { sortBy },
+  } = useTable(
+    {
+      columns,
+      data,
+    },
+    useSortBy,
+  );
+
+  React.useEffect(() => {}, [sortBy]);
```
Comment (ZmicerBoksha) [inline src/components/table/table.tsx:25]: May be you can remove this line

File: src/components/table/table.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  return (
+    <InfiniteScroll dataLength={rows.length} next={update} hasMore={true} loader={<h4>Loading more 2 itens...</h4>}>
+      <table {...getTableProps()}>
+        <thead>
+          {headerGroups.map((headerGroup: any) => (
+            <tr {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map((column: any) => (
+                <th {...column.getHeaderProps(column.getSortByToggleProps())}>
+                  {column.render('Header')}
+                  <span>{column.isSorted ? (column.isSortedDesc ? ' 🔽' : ' 🔼') : ''}</span>
```
Comment (ZmicerBoksha) [inline src/components/table/table.tsx]:
```suggestion
                  <span>{column.isSorted && (column.isSortedDesc ? ' 🔽' : ' 🔼') }</span>
```

**Comment by ZmicerBoksha** [comment]
To many // @ts-ignore, try to get rid of some of them.

File: src/components/candidateTrello/candidateTrello.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import {Grid, Typography} from "@material-ui/core";
+import CandidateMiniCard from "../canditadeMiniCard/candidateMiniCard";
+import React from "react";
+import useAxios from "axios-hooks";
+import {PREFIX} from "../../constants";
+
+type CandidateTrelloProps = {
+    timeZon: string,
+    op?: string
```
Comment (ArteGon) [inline src/components/candidateTrello/candidateTrello.tsx]: what that  variable means?

File: src/components/canditadeMiniCard/candidateMiniCard.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import './candidateMiniCard.scss';
+import {Card, CardContent, Divider, Grid, Paper, Typography} from "@material-ui/core";
+type CandidateMiniCardProps={
+    timeZon:string,
+    op?:string
+}
+const  CandidateMiniCard: React.FC<CandidateMiniCardProps>= ({timeZon,op}) => {
+    const date = new Date("2012-05-24");
```
Comment (ArteGon) [inline src/components/canditadeMiniCard/candidateMiniCard.tsx:8]: Why custom Date === '2012-05-24' ?

File: src/components/navbar/navbar.tsx [hunk 8]
```diff
@@ -175,11 +210,33 @@ export default function Navbar() {
...
         >
           <StaffIcon style={{ marginRight: '5px' }} />
           Staff
         </MenuItem>
+        <Collapse in={staffOpen} timeout="auto" unmountOnExit>
+          <MenuList disablePadding>
+            {['HR', 'TS', 'add'].map((category, index) => {
+              return (
+                <MenuItem
+                  key={index}
```
Comment (ArteGon) [inline src/components/navbar/navbar.tsx:226]: Use index for key is bad practice, can you use any value?

File: src/components/navbar/navbar.tsx [hunk 9]
```diff
@@ -175,11 +210,33 @@ export default function Navbar() {
...
+                  button
+                  className={classes.nested}
+                  component={Link}
+                  to={`/staff/${category}`}
+                  selected={selectedIndex === index + 3}
```
Comment (vchepurnaya) [inline src/components/navbar/navbar.tsx]:
```suggestion
                  selected={selectedIndex === index + 7}
```

File: src/components/navbar/navbar.tsx [hunk 10]
```diff
@@ -175,11 +210,33 @@ export default function Navbar() {
...
+                  onClick={(event: React.MouseEvent<HTMLElement>) => handleListItemClick(event, index + 3)}
```
Comment (vchepurnaya) [inline src/components/navbar/navbar.tsx]:
```suggestion
                  onClick={(event: React.MouseEvent<HTMLElement>) => handleListItemClick(event, index + 7)}
```

**Review by ArteGon — DISMISSED** [review]

File: src/components/candidateTrello/candidateTrello.tsx (hunk 6 above)
Comment (DmitryPateew) [reply src/components/candidateTrello/candidateTrello.tsx]: I removed this variable

File: src/components/canditadeMiniCard/candidateMiniCard.tsx (hunk 7 above)
Comment (DmitryPateew) [reply src/components/canditadeMiniCard/candidateMiniCard.tsx:8]: It is just an example for me how works intl.

**Review by vchepurnaya — APPROVED** [review]

**Review by ZmicerBoksha — APPROVED** [review]

**Review by ArteGon — APPROVED** [review]

**Pull request merged**